_test/test_sqw/worker_4tests_local.m

```diff
@@ -186,44 +198,75 @@
             disp('WORKER_4TESTS:     Entering the job loop ******************')
         end
         
-        % send first "running" log message and set-up starting time.
+        % send first "running" log message and set-up starting time. Runs
+        % asynchronously.
         n_steps = je.n_steps;
         mis.do_logging(0,n_steps);
         %
+        je.do_job_completed = false; % wait at barrier if exception in do_job
         while ~je.is_completed()
             je= je.do_job();
             % when its tested, workers are tested in single Matlab
             % session so it will hand up on synchronization
-            je.do_job_completed = true;
             if ~is_tested
```

Comment:
if we're testing this whole block is disabled so we can never test the is_cancelled logic

Reply:
yes, unfortunately we can not test interprocess communication within single thread. is_tested is set up when tested single-threaded to avoid hanging up on barriers. Herbert parallel job fail-restart is testing this logic in parallel.